akka-remote/src/main/scala/akka/remote/RemoteWatcher.scala

```diff
@@ -177,87 +168,88 @@ private[akka] class RemoteWatcher(
   def quarantine(address: Address, uid: Option[Int]): Unit =
     remoteProvider.quarantine(address, uid)
 
-  def rewatchRemote(watchee: ActorRef, watcher: ActorRef): Unit =
-    if (watching.contains((watchee, watcher)))
-      watchRemote(watchee, watcher)
-    else
-      //has been unwatched inbetween, skip re-watch
-      log.debug("Ignoring re-watch after being unwatched in the meantime: [{} -> {}]", watcher.path, watchee.path)
+  def watchRemote(watchee: InternalActorRef, watcher: InternalActorRef): Unit = {
+    assert(watcher != self)
 
-  def watchRemote(watchee: ActorRef, watcher: ActorRef): Unit =
     if (watchee.path.uid == akka.actor.ActorCell.undefinedUid)
       logActorForDeprecationWarning(watchee)
-    else if (watcher != self) {
-      log.debug("Watching: [{} -> {}]", watcher.path, watchee.path)
-      addWatching(watchee, watcher)
 
-      // also watch from self, to be able to cleanup on termination of the watchee
-      context watch watchee
-      watching += ((watchee, self))
-    }
+    log.debug("Watching: [{} -> {}]", watcher.path, watchee.path)
+    watching.addBinding(watchee, watcher)
+    watchNode(watchee)
```

Comment:
yes, it was, but previously the watch was also sent from RemoteActorRefProvider.afterSendSystemMessage, which is now not done.

I think it was cautious with those actorFor ActorRefs here because they don't include uid and therefore the equals would not match when there is a mix of those and ordinary ActorRef.
I think we should try to retain the old behaviour.